### PR TITLE
[fbgemm ] updated port to latest version

### PR DESCRIPTION
**Pull request by gvavvari**
**Describe the pull request**
This pull request updates fbgemm port

- #### What does your PR fix?
This PR resolves issues mentioned in https://github.com/pytorch/FBGEMM/issues/1070. But the asmjit port needs to be updated as well. 

- #### Does your PR follow the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)?
Yes

### CODE REVIEW

File: ports/fbgemm/portfile.cmake [hunk 1]
```diff
@@ -4,8 +4,8 @@ vcpkg_find_acquire_program(PYTHON3)
 vcpkg_from_github(
     OUT_SOURCE_PATH SOURCE_PATH
     REPO pytorch/fbgemm
-    REF 51bc2ecc7086980fd9e6190163b04486cd37ac2a
-    SHA512 43e59ff25ec9365061be6a8db7ec592f096e74897e338dd3e17401f2188986cf3b7e26f82f7c1e2adf34496fb32cb4b50d40731bee3bf9fa5b1f2ac92a09e671
+    REF 5876514306de2586e617b6ff1212eacf25114a49
```
Comment (JonLiu1993) [inline ports/fbgemm/portfile.cmake:7]: @gvavvari, Thanks for your pr, please update fbgemm to the stable version v0.2.0

File: ports/fbgemm/vcpkg.json [hunk 2]
```diff
@@ -1,7 +1,7 @@
 {
   "name": "fbgemm",
-  "version-date": "2021-03-18",
-  "port-version": 4,
+  "version-date": "2022-09-21",
+  "port-version": 5,
```
Comment (JonLiu1993) [inline ports/fbgemm/vcpkg.json:4]:
```suggestion
```

Comment (JonLiu1993) [reply ports/fbgemm/vcpkg.json:4]: When you update port, you should remove `port-version`

**Review by JonLiu1993 — CHANGES_REQUESTED** [review]

**Comment by gvavvari** [comment]
@JonLiu1993 made changes to portfiles to match ver 0.2.0

File: ports/fbgemm/vcpkg.json [hunk 3]
```diff
@@ -1,7 +1,6 @@
 {
   "name": "fbgemm",
-  "version-date": "2021-03-18",
-  "port-version": 4,
+  "version-date": "2022-07-20",
```
Comment (JonLiu1993) [inline ports/fbgemm/vcpkg.json:3]:
```suggestion
  "version": "0.2.0",
```

**Review by JonLiu1993 — CHANGES_REQUESTED** [review]

**Comment by gvavvari** [comment]
@JonLiu1993 any update on this pull request ?

**Comment by JonLiu1993** [comment]
@gvavvari , I help you formatting the vcpkg.json file, but the ci looks tested failed on many triplet.
This is error log, please take a look:
````
CMake Error at /agent/_work/1/s/scripts/buildsystems/vcpkg.cmake:622 (_add_library):
  _add_library cannot create target "fbgemm" because another target with the
  same name already exists.  The existing target is a static library created
  in source directory
  "/mnt/vcpkg-ci/buildtrees/fbgemm/src/b134ffa3b9-dc570771d5.clean".  See
  documentation for policy CMP0002 for more details.
Call Stack (most recent call first):
  CMakeLists.txt:229 (add_library)
````

File: ports/fbgemm/fix-cmakelists.patch [hunk 4]
```diff
@@ -90,32 +90,27 @@ index 6f19a16..3c758d2 100644
...
 -    $<TARGET_OBJECTS:fbgemm_avx512>)
 -elseif(FBGEMM_LIBRARY_TYPE STREQUAL "shared")
--  add_library(fbgemm SHARED
--    $<TARGET_OBJECTS:fbgemm_generic>
--    $<TARGET_OBJECTS:fbgemm_avx2>
--    $<TARGET_OBJECTS:fbgemm_avx512>)
 +)
 +target_link_libraries(fbgemm_avx512 PUBLIC asmjit::asmjit cpuinfo::cpuinfo)
 +
 +add_library(fbgemm
```
Comment (dg0yt) [inline ports/fbgemm/fix-cmakelists.patch:96]: fbgemm is created here (implicitly using the `BUILD_SHARED_LIBS` setting...

File: ports/fbgemm/fix-cmakelists.patch [hunk 5]
```diff
@@ -90,32 +90,27 @@ index 6f19a16..3c758d2 100644
...
 +  $<TARGET_OBJECTS:fbgemm_generic>
 +  $<TARGET_OBJECTS:fbgemm_avx2>
 +  $<TARGET_OBJECTS:fbgemm_avx512>)
-+if(BUILD_SHARED_LIBS)
-   set_property(TARGET fbgemm_generic PROPERTY POSITION_INDEPENDENT_CODE ON)
-   set_property(TARGET fbgemm_avx2 PROPERTY POSITION_INDEPENDENT_CODE ON)
++if(FBGEMM_LIBRARY_TYPE STREQUAL "shared")
+   add_library(fbgemm SHARED
```
Comment (dg0yt) [inline ports/fbgemm/fix-cmakelists.patch:101]: ... and here again, for shared libs.

**Review by dg0yt — COMMENTED** [review]
You need to review the patching for control over static vs. shared lib.

**Comment by JonLiu1993** [comment]
@gvavvari ,Could you consider the @dg0yt 's suggesstion?

**Comment by JonLiu1993** [comment]
Closing this PR since it seems that no progress is being made. Please ping us to reopen if work is still being done.

**Pull request closed without merging**